library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -353,6 +362,32 @@ public Slice get(Slice key, long timeBucket, ReadSource readSource)
       }
     }
 
+
+    private int filteredCount = 0;
+    private int unfilteredCount = 0;
+    /**
+     * Test the result of using bloom filter and remove it if it not helpful for improving the performance.
+     *
+     * @param mightContain true if collection might contain the value
+     */
+    private void verifyBloomFilter(boolean mightContain)
+    {
+      if (!mightContain) {
+        filteredCount++;
+      } else {
+        unfilteredCount++;
+      }
+
+      if (unfilteredCount > filteredCount * 4) {
+        unloadBloomFilter();
```

Comment:
agree. This logic is based on statistics, so the base number should be relative big. I'll add logic to check base number.